packages/@webex/plugin-meetings/src/statsAnalyzer/index.ts

```diff
@@ -556,253 +494,260 @@ export class StatsAnalyzer extends EventsScope {
+          return;
+        }
+
+        if (this.meetingMediaStatus.expected.sendAudio && mediaType.includes('audio-send')) {
+          currentStats = this.statsResults[mediaType].send;
+          previousStats = this.lastStatsResults[mediaType].send;
 
-        if (
-          currentStats.totalPacketsSent === previousStats.totalPacketsSent ||
-          currentStats.totalPacketsSent === 0
-        ) {
-          LoggerProxy.logger.info(
-            `StatsAnalyzer:index#compareLastStatsResult --> No ${mediaType} RTP packets sent`
-          );
-        } else {
           if (
-            currentStats.totalAudioEnergy === previousStats.totalAudioEnergy ||
-            currentStats.totalAudioEnergy === 0
+            currentStats.totalPacketsSent === previousStats.totalPacketsSent ||
+            currentStats.totalPacketsSent === 0
           ) {
             LoggerProxy.logger.info(
-              `StatsAnalyzer:index#compareLastStatsResult --> No ${mediaType} Energy present`
+              `StatsAnalyzer:index#compareLastStatsResult --> No ${mediaType} RTP packets sent`
             );
+          } else {
+            if (
+              currentStats.totalAudioEnergy === previousStats.totalAudioEnergy ||
+              currentStats.totalAudioEnergy === 0
+            ) {
+              LoggerProxy.logger.info(
+                `StatsAnalyzer:index#compareLastStatsResult --> No ${mediaType} Energy present`
+              );
+            }
+
+            if (currentStats.audioLevel === 0) {
+              LoggerProxy.logger.info(
+                `StatsAnalyzer:index#compareLastStatsResult --> ${mediaType} level is 0 for the user`
+              );
+            }
           }
 
-          if (currentStats.audioLevel === 0) {
-            LoggerProxy.logger.info(
-              `StatsAnalyzer:index#compareLastStatsResult --> ${mediaType} level is 0 for the user`
-            );
-          }
-        }
-
-        this.emitStartStopEvents(
-          mediaType,
-          previousStats.totalPacketsSent,
-          currentStats.totalPacketsSent,
-          true
-        );
-      }
-
-      if (this.meetingMediaStatus.expected.receiveAudio) {
-        // compare audio stats received
-        currentStats = this.statsResults[mediaType].recv;
-        previousStats = this.lastStatsResults[mediaType].recv;
-
-        if (
-          currentStats.totalPacketsReceived === previousStats.totalPacketsReceived ||
-          currentStats.totalPacketsReceived === 0
-        ) {
-          LoggerProxy.logger.info(
-            `StatsAnalyzer:index#compareLastStatsResult --> No ${mediaType} RTP packets received`
-          );
-        } else if (
-          currentStats.totalSamplesReceived === previousStats.totalSamplesReceived ||
-          currentStats.totalSamplesReceived === 0
-        ) {
-          LoggerProxy.logger.info(
-            `StatsAnalyzer:index#compareLastStatsResult --> No ${mediaType} samples received`
+          this.emitStartStopEvents(
```

Comment:
in this call (and all the other ones to emitStartStopEvents()) the mediaType will now be different to what it used to be even for transcoded meetings and as a result we will send call analyzer metrics "client.media.rx.stop" and "client.media.rx.start" with a different mediaType and that is not allowed according to this: https://sqbu-github.cisco.com/WebExSquared/event-dictionary/blob/ba25c9ff3c7440bb423a75ea28c6d0b023833a24/diagnostic-events.raml#L145

Reply:
updated to use a cumulative approach on media types.